docs/releases/3.1.txt

```diff
@@ -67,16 +67,17 @@ It should have no noticeable effect on any existing Django projects.
 JSONField for all supported database backends
 ---------------------------------------------
 
-Django now includes a :class:`models.JSONField <django.db.models.JSONField>`
-and a :class:`forms.JSONField <django.forms.JSONField>` that can be used on all
+Django now includes the :class:`.models.JSONField` and
+:class:`forms.JSONField <django.forms.JSONField>` that can be used on all
 supported database backends. Both fields support the use of custom JSON
 encoders and decoders. The model field supports the introspection, lookups, and
 transforms that were previously PostgreSQL-only.
 
-If your project uses :class:`django.contrib.postgres.fields.JSONField`, plus
-the related form field and transforms, you should adjust to use the new field
-and create a migration. For now, the old fields and transforms are left as a
-reference to the new ones and are deprecated as of this release.
+If your project uses ``django.contrib.postgres.fields.JSONField``, plus the
+related form field and transforms, you should adjust to use the new fields,
+and generate and apply a database migration. For now, the old fields and
```

Comment:
By `generate`, do you mean running `makemigrations`? That would create unnecessary (and problematic) migrations, wouldn't it? Shouldn't the instructions be to "modify existing migrations so they point to `models.JSONField`"?

Reply:
> By `generate`, do you mean running `makemigrations`?

Yes. 

> That would create unnecessary (and problematic) migrations, wouldn't it?

It will create a necessary migration to match a model state. It won't be problematic. 

> Shouldn't the instructions be to "modify existing migrations so they point to `models.JSONField`"?

No.

Reply:
> Shouldn't the instructions be to "modify existing migrations so they point to models.JSONField"?

I think they should. In django 4.0, the postgres JSONField will be dropped, so any historical migrations will fail with `ImportError` at that point, and will need updating to point to the new field anyway.

Reply:
> It will create a necessary migration to match a model state. It won't be problematic.

Ah, yes I forgot about [this](https://github.com/django/django/pull/12392#issuecomment-583797410). Still, what kind of situation would `makemigrations` produce a migration in this case, though? If users are just moving from `contrib.postgres.fields.JSONField` to `models.JSONField`, `makemigrations` won't produce any migrations... (unless they change something else in the model, but that would be unrelated).

> No.

Later on, when `contrib.postgres.fields.JSONField` is removed, an `ImportError` would happen if users still have `contrib.postgres.fields.JSONField` imports in their old migrations. What should they do in that case?

Reply:
> Still, what kind of situation would `makemigrations` produce a migration in this case, though? If users are just moving from `contrib.postgres.fields.JSONField` to `models.JSONField`, `makemigrations` won't produce any migrations...

It will create a migration. Have you checked this? 

> Later on, when `contrib.postgres.fields.JSONField` is removed, an `ImportError` would happen if users still have `contrib.postgres.fields.JSONField` imports in their old migrations. What should they do in that case?
> 
> Edit: ninja'd by @adamchainz

We can support it in historical migrations (like we did in the past with other fields).

Reply:
> We can support it in historical migrations (like we did in the past with other fields).

I will amend it later.

Reply:
> It will create a migration. Have you checked this?

Ah, correct. It didn't the last time I checked, but maybe I was doing something wrong. I just tried again and it did create a migration. Cheers :v:

Reply:
Amended.